src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/AspNet/AspNetRequestStatusTriggerSettings.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet
+{
+    internal sealed class AspNetRequestStatusTriggerSettings : AspNetTriggerSettings
+    {
+        /// <summary>
+        /// Specifies the set of status codes for the trigger.
+        /// E.g. 200-200;400-500
+        /// </summary>
+        [Required]
+        [CustomValidation(typeof(StatusCodeRangeValidator), nameof(StatusCodeRangeValidator.ValidateStatusCodes))]
+        public StatusCodeRange[] StatusCodes { get; set; }
+    }
+
+    internal struct StatusCodeRange
+    {
+        public StatusCodeRange(int min) : this(min, min) { }
+
+        public StatusCodeRange(int min, int max)
+        {
+            Min = min;
+            Max = max;
+        }
+
+        public int Min { get; set; }
+        public int Max { get; set; }
+    }
+
+    public static class StatusCodeRangeValidator
+    {
+        public static ValidationResult ValidateStatusCodes(object statusCodes)
+        {
+            StatusCodeRange[] statusCodeRanges = (StatusCodeRange[])statusCodes;
+            Func<string[]> validationMembers = () => new[] { nameof(AspNetRequestStatusTriggerSettings.StatusCodes) };
```

Comment:
Why isn't this just a readonly field on the class?

Reply:
It doesn't really need to exist unless there's validation failures but I can change it if it's easier to read.